lib/hotelblock.rb

```diff
@@ -0,0 +1,76 @@
+
+require_relative "room"
+require_relative "csv_record"
+
+module HotelGroup
+  class HotelBlock < CsvRecord
+    attr_reader :discount, :id, :start_time, :end_time
+
+    attr_accessor :rooms
+
+    def initialize(id, start_time, end_time, rooms, discount)
+      @discount = discount
+      @id = id
+      @start_time = start_time
+      @end_time = end_time
+
+      if rooms.count > 5
+        raise ArgumentError, "Maximum of 5 rooms allowed in a block"
```

Comment:
This doesn't get tested in your tests! If you're going to raise an error, make sure that you test it!